lib/phoenix/endpoint/supervisor.ex

```diff
@@ -6,7 +6,7 @@ defmodule Phoenix.Endpoint.Supervisor do
   use Supervisor
 
   def start_link(otp_app, endpoint) do
-    if endpoint.config(:server) do
+    if endpoint.config(:server) || endpoint.config(:pubsub)[:adapter] do
```

Comment:
I have renamed this module to `server.ex` in master exactly predicting the addition of the pubsub. So I would move the pubsub out of here and add it directly to the endpoint supervision tree (which we would change to be one for one too).